src/app/tabs/camera.cpp

```diff
@@ -0,0 +1,143 @@
+#include <app/config.hpp>
+#include <app/tabs/camera.hpp>
+#include <app/window.hpp>
+#include <app/theme.hpp>
+
+CameraTab::CameraTab(QWidget *parent) : QWidget(parent)
+{
+    QStackedLayout *stack = new QStackedLayout(this);
+    stack->addWidget(connect_widget());
+    stack->addWidget(cam_widget());
+
+    connect(this, &CameraTab::stream_connected, this, [stack]() {stack->setCurrentIndex(1);} );
+    connect(this, &CameraTab::stream_disconnected, this, [stack]() {stack->setCurrentIndex(0);} );
+
+    player = NULL;
+    // connect_stream(); TODO: offer auto-connect option
+}
+
+QWidget* CameraTab::cam_widget()
+{
+    QWidget* widget = new QWidget(this);
+    QVBoxLayout *layout = new QVBoxLayout(widget);
+
+    QHBoxLayout *connection = new QHBoxLayout;
+    status = new QLabel("Disconnected", widget);
+    status->setFont(Theme::font_16);
+    QPushButton *disconnect = new QPushButton("disconnect");
+    disconnect->setFont(Theme::font_14);
+    disconnect->setFlat(true);
+    disconnect->setIconSize(Theme::icon_36);
+    Theme::get_instance()->add_button_icon("wifi", disconnect);
+    connection->addWidget(status);
+    connection->addStretch();
+    connection->addWidget(disconnect);
+
+    connect(disconnect, &QPushButton::clicked, this, &CameraTab::disconnect_stream);
+
+    video_widget = new QVideoWidget;
+
+    layout->addWidget(video_widget);
+    layout->addLayout(connection);
+
+    return widget;
+}
+
+
+QWidget *CameraTab::connect_widget()
+{
+    QWidget *widget = new QWidget(this);
+    QVBoxLayout *layout = new QVBoxLayout(widget);
+
+    QLabel *label = new QLabel("connect camera stream for camera view", widget);
+    label->setFont(Theme::font_16);
+    label->setAlignment(Qt::AlignCenter);
+
+    Config *config = Config::get_instance();
+    QLineEdit *stream_url_input = new QLineEdit(config->get_cam_stream_url(), widget);
+    stream_url_input->setFixedWidth( 18*30 );
+    stream_url_input->setFont(Theme::font_18);
+    stream_url_input->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+    connect(stream_url_input, &QLineEdit::editingFinished, this, [config,stream_url_input]() {config->set_cam_stream_url(stream_url_input->text());});
+
+    status2 = new QLabel("", widget);
+    status2->setFont(Theme::font_16);
+    status2->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Fixed);
+
+    QPushButton *button = new QPushButton("connect", widget);
+    button->setSizePolicy(QSizePolicy::Fixed, QSizePolicy::Preferred);
+    button->setFont(Theme::font_14);
+    button->setFlat(true);
+    button->setIconSize(Theme::icon_36);
+    Theme::get_instance()->add_button_icon("wifi", button);
+    connect(button, &QPushButton::clicked, this, [this, stream_url_input](){this->connect_stream(stream_url_input->text());});
+
+    connect(this, &CameraTab::stream_connecting, this, [button, stream_url_input]() {button->setEnabled(false); stream_url_input->setEnabled(false);} );

```

Comment:
spaces inbetween brackets (same for the one below)

Reply:
and no extra spaces around the parens

Reply:
also i would recommend breaking up the lambda onto multiple lines to help with the readability